presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeSelector.java

```diff
@@ -44,7 +44,12 @@
      * If we cannot find an assignment for a split, it is not included in the map. Also returns a future indicating when
      * to reattempt scheduling of this batch of splits, if some of them could not be scheduled.
      */
-    SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks);
+    SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks, int limit);
+
+    default SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks)
```

Comment:
This is used only by tests. Could you please migrate the tests instead of leaving it in the interface?

Reply:
wil do.